js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -2073,42 +2075,50 @@ const EditingController = modules.ViewController.inherit((function() {
             const oldEditRowIndex = that._getVisibleEditRowIndex();
 
             if(!isRowEditMode(that)) {
-                result = Deferred();
-                this.executeOperation(result, () => {
-                    this._closeEditCellCore(isError, oldEditRowIndex, withoutSaveEditData);
-                    result.resolve();
+                const deferred = new Deferred();
+                result = new Deferred();
+                this.executeOperation(deferred, () => {
+                    this._closeEditCellCore(isError, oldEditRowIndex, withoutSaveEditData).always(result.resolve);
                 });
             }
+
             return result.promise();
         },
 
         _closeEditCellCore(isError, oldEditRowIndex, withoutSaveEditData) {
-            const that = this;
-            const editMode = getEditMode(that);
-            const dataController = that._dataController;
+            const editMode = getEditMode(this);
+            const dataController = this._dataController;
+            const deferred = new Deferred();
+            const promise = deferred.promise();
```

Comment:
you can not create this variable and return `deferred.promise()`. It's definition seems like something happens with `promise` in code but it is only returned.

Reply:
I did it to prevent redundant lines of code with "deferred.resolve" in different if...else conditions.

Reply:
Ah, sorry, didn't see 2103 line